lib/server/services/shift/ShiftService.js

```diff
@@ -58,40 +58,43 @@ class ShiftService {
      * @param {Date} shiftStart the start time of the shift (to check for previous information)
      * @return {Promise<object>} the shift data
      */
-    async getShiftData(shifterUserIdentifier, shiftType, shiftStart = null) {
+    async getShiftData(shifterUserIdentifier, shiftType) {
         const shift = await this.getUserPendingShiftOrFail(shifterUserIdentifier);
 
         const ret = {
             shift,
             issuesLogs: await this.getShiftIssues(shift.start, shiftType),
             typeSpecific: null,
-            previousInfo: { value: '' },
+            infoFromPreviousShifter: { value: '' },
         };
 
         // Find EoS reports from the last shift with the correct title
-        const timestamp = shiftStart || shift.start;
-        const pastShift = getShiftFromTimestamp(timestamp - SHIFT_DURATION);
+        const pastShift = getShiftFromTimestamp(shift.start - SHIFT_DURATION);
         const title = formatEosReportTitle(pastShift, shiftType);
 
-        const prevShiftIssues = await getShiftIssuesByTitle(pastShift, title);
+        const prevShiftIssues = await getShiftIssuesByTitle(title);
 
-        if (prevShiftIssues.length !== 1) {
-            ret.previousInfo.errorMessage = `No ${shiftType} EoS reports found from the previous shift`;
+        if (prevShiftIssues.length === 0) {
+            ret.infoFromPreviousShifter.errorMessage = `No ${shiftType} EoS reports found from the previous shift`;
+        } else if (prevShiftIssues.length > 1) {
+            ret.infoFromPreviousShifter.errorMessage = `Multiple ${shiftType} EoS reports found from the previous shift`;
         } else {
             const eosReportText = prevShiftIssues[0].text;
             if (!eosReportText) {
-                ret.previousInfo.errorMessage = 'EoS report from the previous shift is missing the text field';
+                ret.infoFromPreviousShifter.errorMessage = 'EoS report from the previous shift is missing the text field';
             } else {
-                const infoFromPreviousShifterRegex = /(?<=\n\n### For next shifter\n)([\S\s]*)(?=\n\n###)/g;
+                const infoFromPreviousShifterRegex = /(?<=\n\n### For next shifter\n)([\S\s]*?)(?=\n\n###)/g;
```

Comment:
`*` is already allows for `0` match, I think `?` is not necessary here

Reply:
The ? makes it a lazy operator so that I can test for duplicate fields (the tests don't work without it)